src/cargo/ops/cargo_output_metadata.rs

```diff
@@ -22,16 +23,34 @@ pub struct OutputMetadataOptions<'a> {
 /// used versions - considering overrides - and writes all dependencies in a JSON
 /// format to stdout.
 pub fn output_metadata(opt: OutputMetadataOptions, config: &Config) -> CargoResult<ExportInfo> {
+    assert_eq!(opt.version, VERSION);
```

Comment:
Oh actually, while you're at it, can you change this assertion to a nicer error? For example something along the lines of:

```rust
if opt.version != VERSION {
    bail!("metadata version {} not supported, only {} is currently supported", opt.version, VERSION);
}
```